### PR TITLE
Core/SAI: Add SMART_ACTION_ACTIVATE_GAMEOBJECT action

### LINKED ISSUE #27196: Core/SAI: SMART_ACTION_ACTIVATE_GAMEOBJECT
### Description

GameObjectActions implemented in https://github.com/TrinityCore/TrinityCore/pull/24997 originally are supposed to be used in several systems, as stated in https://github.com/TrinityCore/TrinityCore/pull/24997#issuecomment-657087699

More info here https://github.com/TrinityCore/TrinityCore/issues/27009#issuecomment-932972363

### Expected behaviour

Allow to use GameObjectActions in SAI by SMART_ACTION_ACTIVATE_GAMEOBJECT with param1 = GameObjectActions
And in C++ scripts by calling ActivateGameObject

### Steps to reproduce the problem

none

### Branch

3.3.5

### TC rev. hash/commit

222164a5cc11b4c012074921d4ae1ae74276a990

### Operating system

Win 8.1

### Custom changes

none

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Add SMART_ACTION_ACTIVATE_GAMEOBJECT action
- Move current implementation of ActivateObject from Spell to GameObject

**Issues addressed:**

Closes #27196

**Tests performed:**

None

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObjectData.h [hunk 1]
```diff
@@ -699,6 +699,7 @@ enum class GameObjectActions : uint32
     UseArtKit2,                     // Use ArtKit2
     UseArtKit3,                     // Use ArtKit3
     SetTapList,                     // Set Tap List
+    Max = SetTapList
```
Comment (Shauren) [inline src/server/game/Entities/GameObject/GameObjectData.h:702]: Why? the entire point of not assigning a value is to have it update automatically

**Comment by offl** [comment]
Errors are outdated a little because actions are not used anymore only for spells

File: src/server/game/Entities/GameObject/GameObjectData.h (hunk 1 above)
Comment (jackpoz) [reply src/server/game/Entities/GameObject/GameObjectData.h:702]: And how do you know which value is max from cpp ? Checking everything the header to make sure it's the latest ? And what if master branch gets more enum values and SetTapList is not the max anymore ? Will you check and update every place that uses SetTapList ?

Comment (Shauren) [reply src/server/game/Entities/GameObject/GameObjectData.h:702]: What I mean is not explicitly assigning a value to `Max`, not "not defining Max at all"

Comment (jackpoz) [reply src/server/game/Entities/GameObject/GameObjectData.h:702]: ah ok

File: src/server/game/AI/SmartScripts/SmartScriptMgr.cpp [hunk 2]
```diff
@@ -2123,6 +2124,19 @@ bool SmartAIMgr::IsEventValid(SmartScriptHolder& e)
...
         }
+        case SMART_ACTION_ACTIVATE_GAMEOBJECT:
+        {
+            if (!NotNULL(e, e.action.activateGameObject.gameObjectAction))
+                return false;
+
+            if (e.action.activateGameObject.gameObjectAction >= uint32(GameObjectActions::Max))
+            {
+                TC_LOG_ERROR("sql.sql", "SmartAIMgr: Entry %d SourceType %u Event %u Action %u has gameObjectAction parameter out of range (max allowed %u, current value %u), skipped.",
+                    e.entryOrGuid, e.GetScriptType(), e.event_id, e.GetActionType(), uint32(GameObjectActions::Max), e.action.activateGameObject.gameObjectAction);
```
Comment (jackpoz) [inline src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:2135]: Max allowed is Max - 1 here, not Max

Comment (offl) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:2135]: Why it's - 1?

Comment (offl) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:2135]: Yeah, I see
I forgot I removed your Max = SetTapList
Then uint32(GameObjectActions::SetTapList) or it's not what you wanted for both branches?

Comment (jackpoz) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:2135]: We should still use Max, just writing `uint32(Max) - 1` in the log.
I'll change it when I find time :)

**Pull request merged**